common/rpc/interceptor/namespace_validator.go

```diff
@@ -128,12 +137,16 @@ func (ni *NamespaceValidatorInterceptor) NamespaceValidateIntercept(
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
-	err := ni.setNamespaceIfNotPresent(req)
-	if err != nil {
-		return nil, err
-	}
 	reqWithNamespace, hasNamespace := req.(NamespaceNameGetter)
 	if hasNamespace {
+
+		if reqWithNamespace.GetNamespace() == "" {
+			err := errNamespaceNotSet
+			metrics.ServiceErrNamespaceNotSetCounter.With(ni.metricsHandler).Record(1)
+			ni.logger.Error("namespace not provided for request", tag.Error(err))
+			return nil, err
```

Comment:
Shall we have a feature flag for controlling the behavior of setting Namespace or not?

Log and metric will always be emitted so that we can know if any customer will be impacted before stop setting Namespace field.

Reply:
It might be useful for you to go read the internal ticket that prompted this work. Yimin proposed the dynamic config initially, but then left another message which seemed to imply we could skip that. I may also be misunderstanding his messages and he wants to do all these things with time/deployments between the changes.